server/src/main/java/org/apache/iotdb/db/writelog/io/SingleFileLogReader.java

```diff
@@ -53,54 +54,69 @@
 
   private boolean fileCorrupted = false;
 
+  private boolean hasNext = false;
+
   public SingleFileLogReader(File logFile) throws FileNotFoundException {
     open(logFile);
   }
 
   @Override
   public boolean hasNext() {
-    try {
-      if (batchLogReader != null && batchLogReader.hasNext()) {
-        return true;
-      }
-
-      if (logStream.available() < LEAST_LOG_SIZE) {
-        return false;
-      }
 
-      int logSize = logStream.readInt();
-      if (logSize <= 0) {
-        return false;
-      }
-      buffer = new byte[logSize];
+    if (Objects.nonNull(batchLogReader)) {
```

Comment:
IMO,   keep `if (batchLogReader != null && batchLogReader.hasNext())` is better...

Reply:
Yes,  if `hasNext()` should be called, there is not need to make any changes here